alerter.c

```diff
@@ -3,29 +3,72 @@
 
 int alertFailureCount = 0;
 
+typedef enum
+{
+   Production,
+   Test
+}Environment;
+
 int networkAlertStub(float celcius) {
     printf("ALERT: Temperature is %.1f celcius.\n", celcius);
-    // Return 200 for ok
-    // Return 500 for not-ok
-    // stub always succeeds and returns 200
-    return 200;
+    if(celcius<=200)
+    {
+     return 200;
+     }
+    else
+    {
+     return 500;
+    }
 }
 
-void alertInCelcius(float farenheit) {
+int networkAlertProduction(float celcius){
+  if(celcius<=200)
+  {
+     return 200;
+  }
+  else
+  {
+    return 500;
+  }
+}  
+
+float SelectAlertEnvironment(float celcius, Environment environment_e)
+{
+    float calculated_Value;
+   if(environment_e == Production)
+   {
+     calculated_Value = networkAlertProduction(celcius);
+   }
+   else
+   {
+     calculated_Value = networkAlertStub(celcius);
+   }
+   return calculated_Value;
+}
+    
+float convertfarenheittocelcius(float farenheit)
+{
     float celcius = (farenheit - 32) * 5 / 9;
-    int returnCode = networkAlertStub(celcius);
+    return celcius;
+}
+
+void alertInCelcius(float farenheit,Environment environment_e) {
+    float celcius  = convertfarenheittocelcius(farenheit);
+    int returnCode = SelectAlertEnvironment(celcius,environment_e);

```

Comment:
Boolean value is used to select the environment. This code can be erroneous  if somehow non production value is passed in the production environment.
Best way is to abstract using interface or function pointer and inject based on the environment.
Production code need not be aware of test or production envt.